lib/dynflow/persistence_adapters/sequel.rb

```diff
@@ -242,22 +252,37 @@ def migrate_db
         ::Sequel::Migrator.run(db, self.class.migrations_path, table: 'dynflow_schema_info')
       end
 
-      def prepare_record(table_name, value, base = {})
+      def prepare_record(table_name, value, base = {}, with_data = true)
         record = base.dup
-        if table(table_name).columns.include?(:data)
+        if with_data && table(table_name).columns.include?(:data)
           record[:data] = dump_data(value)
+        else
+          record[:data] = nil
+          record.merge! serialize_columns(table_name, value)
         end
+
         record.merge! extract_metadata(table_name, value)
         record.each { |k, v| record[k] = v.to_s if v.is_a? Symbol }
+
         record
       end
 
-      def save(what, condition, value)
+      def serialize_columns(table_name, record)
+        record.reduce({}) do |acc, (key, value)|
+          if SERIALIZABLE_COLUMNS.fetch(table_name, []).include?(key.to_s)
+            acc.merge(key.to_sym => dump_data(value))
+          else
+            acc
+          end
+        end
+      end
+
+      def save(what, condition, value, with_data = true)
```

Comment:
Since `with_data` is used only for testing of the legacy approach, would it be possible to move the preparation of the test data to the test itself, so that the persistence code itself doesn't have to care about saving anything to data from now on?

Reply:
It is still used by at least coordinator records. I'll check whether we could do something about it

Reply:
Oh, in that case that's not that bad. I'm ok with that unless there is some better way.

Reply:
I can't think of a solution which would make this significantly better. Let's keep it the way it is for now